src/scene/minigame_scenes/mgscene_penguin.cpp

```diff
@@ -0,0 +1,108 @@
+#include "mgscene_penguin.h"
+#include "game_objects/player.h"
+#include "game_objects/minigame/empty_collider.h"
+#include "events.h"
+#include "util/color.h"
+#include "../../asset_loader.h"
+
+void MGScenePenguin::s_update() {
+    Scene::s_update();
+}
+
+void MGScenePenguin::c_update() {
+    Scene::c_update();
+}
+
+void MGScenePenguin::s_setup() {
+    // Set up penguins. Initially disabled.
+    for (int i = 0; i < MAX_PENGUINS; i++) {
+        Penguin* to_add = new Penguin();
```

Comment:
For future reference, please use smart pointers (`std::unique_ptr` and `std::shared_ptr`) so we don't have to worry about freeing memory.